tasks/tasks.py

```diff
@@ -5,61 +5,343 @@
+
+    The behavior is equivalent to `$ python export_db.py --jobs`, which lives in main function of
+    export_db.py.
+    """
+    def __init__(self, client_config, args):
+        super().__init__(name="export_jobs")
+        self.client_config = client_config
+        self.args = args
+
+    def run(self):
+        jobs_c = JobsClient(self.client_config)
+        jobs_c.log_job_configs()
+
+class JobsImportTask(AbstractTask):
+    """Task that imports all jobs.
+
+    The behavior is equivalent to `$ python import_db.py --jobs`, which lives in main function of
     import_db.py.
     """
+    def __init__(self, client_config, args):
+        super().__init__(name="import_jobs")
+        self.client_config = client_config
+        self.args = args
+
+    def run(self):
+        jobs_c = JobsClient(self.client_config)
+        jobs_c.import_job_configs()
+
+class MetastoreExportTask(AbstractTask):
+    """Task that exports all metastore tables.
 
+    The behavior is equivalent to `$ python export_db.py --metastore`, which lives in main function of
+    export_db.py.
+    """
+    def __init__(self, client_config, checkpoint_service, args):
+        super().__init__(name="export_metastore")
+        self.client_config = client_config
+        self.checkpoint_service = checkpoint_service
+        self.args = args
+
+    def run(self):
+        hive_c = HiveClient(self.client_config, self.checkpoint_service)
+        hive_c.export_hive_metastore(cluster_name=self.args.cluster_name, has_unicode=self.args.metastore_unicode)
+
+class MetastoreImportTask(AbstractTask):
+    """Task that imports all metastore tables.
+
+    The behavior is equivalent to `$ python import_db.py --metastore`, which lives in main function of
+    import_db.py.
+    """
+    def __init__(self, client_config, checkpoint_service, args):
+        super().__init__(name="import_metastore")
+        self.client_config = client_config
+        self.checkpoint_service = checkpoint_service
+        self.args = args
+
+    def run(self):
+        hive_c = HiveClient(self.client_config, self.checkpoint_service)
+        # log job configs
+        hive_c.import_hive_metastore(cluster_name=self.args.cluster_name, has_unicode=self.args.metastore_unicode,
+                                     should_repair_table=self.args.repair_metastore_tables)
+
+class MetastoreTableACLExportTask(AbstractTask):
+    """Task that exports all metastore table ACLs.
+
+    The behavior is equivalent to `$ python export_db.py --table-acls`, which lives in main function of
+    export_db.py.
+    """
+    def __init__(self, client_config, args):
+        super().__init__(name="export_metastore_table_acls")
+        self.client_config = client_config
+        self.args = args
+
+    def run(self):
+        table_acls_c = TableACLsClient(self.client_config)
+        notebook_exit_value= table_acls_c.export_table_acls(db_name='')
+        if notebook_exit_value['num_errors'] == 0:
```

Comment:
Is it OK for these errors get ignored? If not, it's better to raise exceptions instead of printing.

Reply:
We are planning to revisit the error handling. There are multiple places where the scripts eat errors silently without logging. We can address this at that time. This is currently just a copy of what TableACLs export does in export_db.py.